### PR TITLE
feat: screenshots and auto note cover

### PR DESCRIPTION
**Pull request by slaveeks**
1. Added util for making screenshots of HTML elements
2. Added making note screenshots of note, when note changed
3. Updating note cover on note changed
4. Added `NoteAttachmentService`
5. Added updating cover to `useNoteSettings`

### CODE REVIEW

File: src/application/services/useNoteSettings.ts [hunk 1]
```diff
@@ -146,6 +153,24 @@ export default function (): UseNoteSettingsComposableState {
...
 
+  /**
+   * Update note cover pisture
+   * @param id - note id
+   * @param data - picture binary data
+   */
+  const updateCover = async (id: NoteId, data: Blob): Promise<void> => {
+    const key = await noteAttachmentService.upload(id, data);
+
+    const { cover } = await noteSettingsService.patchNoteSettingsByNoteId(id, { cover: key });
```
Comment (neSpecc) [inline src/application/services/useNoteSettings.ts:164]: it's better to move it to the domain level since it is a business logic

**Review by GeekaN2 — APPROVED** [review]
NICE 🕶

File: src/infrastructure/utils/screenshot.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import html2canvas from 'html2canvas';
+
+/**
+ * Make html element screenshot
+ * @param elementId - id of element, which have to be screenshooted
+ * @returns binary image data
+ */
+export async function makeElementScreenshot(elementId: string): Promise<Blob | null> {
```
Comment (neSpecc) [inline src/infrastructure/utils/screenshot.ts:8]: its better to pass element to not to care about id assigning

File: src/presentation/pages/Note.vue [hunk 3]
```diff
@@ -26,6 +26,7 @@
       :read-only="!canEdit"
       @change="noteChanged"
     />
+    <div id="note-clone" />
```
Comment (neSpecc) [inline src/presentation/pages/Note.vue:29]: seems unexpected, it's better to incapsulate all screenshot logic at the util level

File: src/application/services/useNoteSettings.ts (hunk 1 above)
Comment (slaveeks) [reply src/application/services/useNoteSettings.ts]: done

File: src/infrastructure/utils/screenshot.ts (hunk 2 above)
Comment (slaveeks) [reply src/infrastructure/utils/screenshot.ts]: move a lot of logic to util

File: src/presentation/pages/Note.vue (hunk 3 above)
Comment (slaveeks) [reply src/presentation/pages/Note.vue]: removed

File: src/presentation/pages/Note.vue [hunk 4]
```diff
@@ -79,11 +83,37 @@ const editor = ref<typeof Editor | undefined>(undefined);
...
+  /**
+   * Get html element with note
+   */
+  const editorElement = document.getElementById('editorjs');
+
+  /**
+   * If it does not exist, we can not do screenshot
+   */
+  if (editorElement !== null) {
+    updatedNoteCover = await makeElementScreenshot(editorElement, {
```
Comment (neSpecc) [inline src/presentation/pages/Note.vue:100]: its better to move it inside the `!isEmpty()` statement

Comment (slaveeks) [reply src/presentation/pages/Note.vue]: done

**Review by neSpecc — APPROVED** [review]

**Pull request merged**